Sources/NIOHTTPCompression/HTTPResponseDecompressor.swift

```diff
@@ -68,25 +86,27 @@ public final class NIOHTTPResponseDecompressor: ChannelDuplexHandler, RemovableC
 
             context.fireChannelRead(data)
         case .body(var part):
-            switch self.state {
-            case .compressed(_, let originalLength):
+            switch self.compression {
+            case .some(var compression):
+                compression.compressedLength += part.readableBytes
```

Comment:
This can't be right: we set the `compressedLength` above differently depending on what mode we're in, but we are unconditionally adding to it now. That seems not right.

Reply:
Yeah, this looks like it is just leftover from a previous version of the patch. I think this indicates we need some extra tests to ensure that our compression limits still apply in the face of having no clear length specification here. Mind doing that?